docs/dotnet-configuration.md

```diff
@@ -0,0 +1,141 @@
+---
+id: dotnet-configuration
+title: Scope .NET Agent configuration
+sidebar_label: Configuration
+---
+
+## Adding agent metadata
+
+You can send arbitrary metadata for every test associated with certain commit which will be shown by Scope.
+
+Additionally, it is possible to set an environment variable as metadata value for a certain key, whose final value will be evaluated at runtime.
+
+### Using Environment Variables
+
+```sh
+export SCOPE_METADATA="sample.key1=$SAMPLE_VAR1,sample.key2=$SAMPLE_VAR2,sample.key3=sampleValue3"
+```
+
+### Using YAML Configuration file
+
+```yaml
+scope:
+  metadata:
+    'sample.key1': $SAMPLE_VAR1
+    'sample.key2': $SAMPLE_VAR2
+    'sample.key3': sampleValue3
+```
+
+## Select agent metadata as test configuration
+
+Yon can select metadata keys to be considered as relevant in the configuration of the test.
+
+By default, Scope .NET agent adds both `platform.name` and `dotnet.runtime` metadata key to the test configuration.
+
+### Using Environment Variables
+
+```sh
+export SCOPE_CONFIGURATION="sample.key1,sample.key2"
+```
+
+### Using YAML Configuration file
+
+```yaml
+scope:
+  configuration:
+    - 'sample.key1'
+    - 'sample.key2'
+```
+
+## Adding code path information
+
+You can add code path information to show the executed lines of a certain test in a certain commit.
+
+By default, Scope .NET agent will not show send the code path.
+
+### Using Environment Variables
+
+```sh
+export SCOPE_CODE_COVERAGE=True
+```
+
+### Using YAML Configuration file
+
+```yaml
+scope:
+  instrumentation:
+    testsframeworks:
+      code_coverage: true
```

Comment:
Code coverage can also be activated outside of testing frameworks, right? Why is it inside `testsframeworks`?

Reply:
And why `code_coverage` is separated by an underscore, but `testsframeworks` is not? Seems a bit inconsistent to me.

Reply:
Well I have both, one inside the tests frameworks instrumentation and one inside the http server instrumentation, yeah we can remove it from there and just call it CODEPATH in the yaml and in the env var as SCOPE.CODEPATH

Reply:
Updated on the 0.1.16 release of the dotnet agent.